dfvfs/analyzer/analyzer_helper.py

```diff
@@ -7,10 +7,13 @@
 class AnalyzerHelper(object):
   """Analyzer helper interface."""
 
+  # pylint: disable=missing-raises-doc,redundant-returns-doc
```

Comment:
No, raising NotImplementedError is by design, it is not to provide an abstract property. I can change this to a dfvfs specific exception if that makes it more clear (I would recommend a separate CL) to a dfvfs.NotImplemented or dfvfs.NotSupported.

Reply:
> Please also move the redundant-returns disable to apply to just the analyzefileobject method

No, this the behaviour of this check  is broken. https://github.com/PyCQA/pylint/issues/984

Reply:
> No, raising NotImplementedError is by design, it is not to provide an abstract property. I can change this to a dfvfs specific exception if that makes it more clear (I would recommend a separate CL) to a dfvfs.NotImplemented or dfvfs.NotSupported.

But raising NotImplementedError does define an abstract method - that's what the exception is for, per the python docs.  Changing this to a dfvfs specific error is better, but we're already raising RuntimeError elsewhere to indicate that something isn't implemented: https://github.com/log2timeline/dfvfs/blob/master/dfvfs/analyzer/analyzer_helper.py#L48

> No, this the behaviour of this check is broken. PyCQA/pylint#984
That issue was closed in 2016, what do you mean by this comment? Why would the check being 'broken' mean that we need to disable this for the whole class? Pylint is complaining because AnalyzeFileObject says it returns something when it just raises.

Reply:
> Changing this to a dfvfs specific error is better, but we're already raising RuntimeError elsewhere

Let's unify this in a separate CL.

> That issue was closed in 2016, what do you mean by this comment?

That issue was for pylint 2.0.0 and looks not backported (hard to tell). With 1.7.x I've been running into various non-consistent errors for interface classes with 'redundant-returns', I opt to disable it for the entire class for the immediate future and let's revisit when we're on pylint 2.1.1. Doing is on a per class level it is easier to spot them at a later time

Reply:
It sounds like we need to do some unification of pylint versions too - we're using 1.9.3 (last python 2 version) elsewhere. 

I don't really get what you mean by "easier to spot" - grep works pretty well everywhere, and when things are disabled at the class level, it's hard to work out what exactly is being overridden.